File: component-factory/base/src/main/java/com/lazylite/mod/utils/FontUtils.java
package com.lazylite.mod.utils;

import android.content.Context;
import android.graphics.Typeface;

import com.lazylite.mod.App;

public class FontUtils {

    public static FontUtils fontsUtil;

    private Context mContext;
    private Typeface iconFontTypeface;


    private FontUtils(Context context) {
        this.mContext = context.getApplicationContext();
    }

    public static FontUtils getInstance() {
        if (fontsUtil == null) {
            fontsUtil = new FontUtils(App.getInstance());
        }
        return fontsUtil;
    }

    public Typeface getIconFontType(Context context) {
        if (iconFontTypeface == null) {
            try {
                iconFontTypeface = Typeface.createFromAsset(context.getAssets(), "fonts/iconfont.ttf");
            } catch (Exception e) {

            }
        }
        return iconFontTypeface;
    }


}

File: component-factory/UICompose/src/main/java/com/godq/compose/iconview/FontUtils.kt
package com.godq.compose.iconview

import android.graphics.Typeface
import com.godq.compose.UICompose
import java.lang.Exception

private var iconFontTypeface: Typeface? = null

fun getIconFontType(): Typeface? {
    if (iconFontTypeface == null) {
        val context = UICompose.applicationContext ?: return null
        val path = UICompose.config?.fontAssetsPath ?: return null
        try {
            iconFontTypeface = Typeface.createFromAsset(context.assets, path)
        } catch (e: Exception) {
        }
    }
    return iconFontTypeface
}